### DISCUSSION TITLE
Compilation error for WT32 -SC01 plus  for ESP32 -TUX

### DISCUSSION THREAD

**Discussion by dorel50** [sukesh-ak/ESP32-TUX#36] [General]
FAILED: esp-idf/LovyanGFX/CMakeFiles/__idf_LovyanGFX.dir/src/lgfx/v1/platforms/esp32/common.cpp.obj
esptool.py v4.5.1
Creating esp32s3 image...
Merged 1 ELF section
Successfully created esp32s3 image.
Generated C:/Espressif/frameworks/esp-idf-v5.0.3/ESP32-TUX/build/bootloader/bootloader.bin
[106/106] cmd.exe /C "cd /D C:\Espressif\frameworks\esp-idf-v5.0.3\ESP32-TUX\build\bootloader\esp-idf\esptool_py && C:\Espressif\python_env\idf5.0_py3.11_env\Scripts\python.exe C:/Espressif/frameworks/esp-idf-v5.0.3/components/partition_table/check_sizes.py --offset 0x8000 bootloader 0x0 C:/Espressif/frameworks/esp-idf-v5.0.3/ESP32-TUX/build/bootloader/bootloader.bin"
Bootloader binary size 0x5080 bytes. 0x2f80 bytes (37%) free.
ninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the C:\Espressif\frameworks\esp-idf-v5.0.3\ESP32-TUX\build\log\idf_py_stderr_output_11532 and C:\Espressif\frameworks\esp-idf-v5.0.3\ESP32-TUX\build\log\idf_py_stdout_output_11532
C:\Espressif\frameworks\esp-idf-v5.0.3\ESP32-TUX>
Please hel mee

**Comment by sukesh-ak** [comment]
First line error shows the failure from LovyanGFX. 
I don't think you have gone through the instructions given in the readme here
https://github.com/sukesh-ak/ESP32-TUX

**Comment by Py-Jinx** [comment]
I definitely followed your instructions, but get the same error. And I tried it in Windows (with IDF version 5.0.4) and in a Debian Windows  Subsystem Linux (with IDF version 5.1.2).

Errors start with:

_[793/1183] Building CXX object esp-idf/LovyanGFX/CMakeFiles/__idf_LovyanGFX.dir/src/lgfx/v1/misc/pixelcopy.cpp.objFAILED: esp-idf/LovyanGFX/CMakeFiles/__idf_LovyanGFX.dir/src/lgfx/v1/misc/pixelcopy.cpp.obj_

and end with:

_/home/jan/ESP32-TUX/components/LovyanGFX/src/lgfx/v1/misc/pixelcopy.cpp: In constructor 'lgfx::v1::pixelcopy_t::pixelcopy_t(const void*, lgfx::v1::color_depth_t, lgfx::v1::color_depth_t, bool, const void*, uint32_t)':
/home/jan/ESP32-TUX/components/LovyanGFX/src/lgfx/v1/misc/pixelcopy.cpp:39:25: error: member 'lgfx::v1::pixelcopy_t::<anonymous>' is used uninitialized [-Werror=uninitialized]
   39 |     , src_mask  ( (1 << src_bits) - 1 )
      |                         ^~~~~~~~
/home/jan/ESP32-TUX/components/LovyanGFX/src/lgfx/v1/misc/pixelcopy.cpp:40:25: error: member 'lgfx::v1::pixelcopy_t::<anonymous>' is used uninitialized [-Werror=uninitialized]
   40 |     , dst_mask  ( (1 << dst_bits) - 1 )
      |                         ^~~~~~~~
cc1plus: some warnings being treated as errors
[810/1183] Building CXX object esp-idf/LovyanGFX/CMakeFiles/__idf_LovyanGFX.dir/src/lgfx/v1/LGFXBase.cpp.objninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /home/jan/ESP32-TUX/build/log/idf_py_stderr_output_6114 and /home/jan/ESP32-TUX/build/log/idf_py_stdout_output_6114_

Any ideas how I can overcome these issues?

**Reply by sukesh-ak** [reply]
Will check during weekend and update. 
FYI, I will always be using IDF latest released version only.

**Reply by darius61** [reply]
I have same issue and build errors as shown above using esp-idf 5.1 and following the instructions in video... Thanks so much for looking into this!!!

**Comment by sukesh-ak** [comment]
Compiler error is now fixed and details are [here](https://github.com/sukesh-ak/ESP32-TUX/issues/50)
It's coming from LovyanGFX so just updated it to latest version.

@dorel50 
@Py-Jinx 
@darius61

**Reply by Py-Jinx** [reply]
Thanks Sukesh! Now it compiles correctly (with warnings only).

**Reply by darius61** [reply]
Same here - looks good -thanks!!!
